src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.Methods.cs

```diff
@@ -46,9 +46,9 @@ public static void WriteMethodBodyStatement(this CodeWriter writer, MethodBodySt
                     }
                     break;
                 case IfStatement ifStatement:
-                    writer.WriteMethodBodyStatement(new IfElseStatement(ifStatement.Condition, ifStatement.Body, null));
+                    writer.WriteMethodBodyStatement(new IfElseStatement(ifStatement.Condition, ifStatement.Body, null, ifStatement.Inline, ifStatement.AddBraces));
```

Comment:
It is hacky to derive `IfStatement` from `IfElseStatement`, so I have done it the other way around: made `IfStatement` part of `IfElseStatement`.

Reply:
But I kind of think that IfStatement is a special form of IfElseStatement therefore their relationship is "is a" so that one should inherit from the other.
But anyhow, this works fine